### PR TITLE
wildcard redirect uri content addition

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** wildcard redirect uri content addition
- **Is this PR related to a Monolith release?** Yes 2021.07.0

### Resolves:

* [OKTA-406718](https://oktainc.atlassian.net/browse/OKTA-406718)

Screen Shot 2021-06-24 at 4 27 55 PM

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/create-okta-application/index.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ Before you can sign a user in, you need to create an Okta app integration that r
 1. Enter a name for your app integration (or leave the default value).
 1. Select **Authorization Code** and **Refresh Token** as the **Grant type**. This enables Authorization Code flow with PKCE for your application and the ability to refresh the access token when it expires without prompting the user to re-authenticate.
 1. Enter values for the **Sign-in redirect URIs**. This is the callback from <GuideLink link="../define-callback/">Define a callback route</GuideLink>. Add values for local development (for example, `http://localhost:8080/login/callback` or `http://localhost:8080/authentication/login-callback` for Blazor) and production (for example, `https://app.example.com/login/callback`).
+
+    If your OpenID Connect client is configured to support subdomains and you want to use a single redirect URI with a wildcard, select the **Allow wildcard * in sign-in redirect URI** checkbox.
```
Comment (venkatviswanathan-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-spa/create-okta-application/index.md:19]: If your OpenID Connect client could have multiple redirect URIs and you want to use a single redirect URI with a wildcard for the subdomain, .....

File: packages/@okta/vuepress-site/docs/guides/sign-into-web-app/create-okta-application/index.md [hunk 2]
```diff
@@ -14,6 +14,13 @@ Before you can sign a user in, you need to create an Okta app integration that r
     > **Note:** It is important to choose the appropriate application type for apps that are public clients. Failing to do so may result in Okta API endpoints attempting to verify an app's client secret, which public clients aren't designed to have, and would break the sign-in or sign-out flow.
 1. Enter a name for your app integration (or leave the default value).
 1. Enter values for the **Sign-in redirect URIs**. This is the callback described in <GuideLink link="../define-callback">Understand the callback route</GuideLink>. Add values for local development (for example, `http://localhost:8080/authorization-code/callback`) and production (for example, `https://app.example.com/authorization-code/callback`).
+
+    If your OpenID Connect client is configured to support subdomains and you want to use a single redirect URI with a wildcard, select the **Allow wildcard * in sign-in redirect URI** checkbox.
+
```
Comment (venkatviswanathan-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-web-app/create-okta-application/index.md:19]: Same as above

**Review by venkatviswanathan-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/create-okta-application/index.md [hunk 3]
```diff
@@ -15,6 +15,13 @@ Before you can sign a user in, you need to create an Okta app integration that r
...
 1. Enter values for the **Sign-in redirect URIs**. This is the callback from <GuideLink link="../define-callback/">Define a callback route</GuideLink>. Add values for local development (for example, `http://localhost:8080/login/callback` or `http://localhost:8080/authentication/login-callback` for Blazor) and production (for example, `https://app.example.com/login/callback`).
+
+    If your OpenID Connect client has multiple redirect URIs and you want to use a single redirect URI with a wildcard for the subdomain, select the **Allow wildcard * in sign-in redirect URI** checkbox.
+
+    > **Caution:** The use of wildcard subdomains is discouraged as an insecure practice, since it may allow malicious actors to have tokens or authorization codes sent to unexpected or attacker-controlled pages. Exercise caution if you decide to include a wildcard redirect URI in your configuration.
+
+    See the parameter [Details](https://developer.okta.com/docs/reference/api/apps/#details) section on the Apps API Reference page for configuration guidance on the use of wildcard subdomains.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-spa/create-okta-application/index.md:23]: link should be shortened to (/docs/reference/api/apps/#details)

File: packages/@okta/vuepress-site/docs/guides/sign-into-web-app/create-okta-application/index.md [hunk 4]
```diff
@@ -14,6 +14,13 @@ Before you can sign a user in, you need to create an Okta app integration that r
...
 1. Enter a name for your app integration (or leave the default value).
 1. Enter values for the **Sign-in redirect URIs**. This is the callback described in <GuideLink link="../define-callback">Understand the callback route</GuideLink>. Add values for local development (for example, `http://localhost:8080/authorization-code/callback`) and production (for example, `https://app.example.com/authorization-code/callback`).
+
+    If your OpenID Connect client has multiple redirect URIs and you want to use a single redirect URI with a wildcard for the subdomain, select the **Allow wildcard * in sign-in redirect URI** checkbox.
+
+    > **Caution:** The use of wildcard subdomains is discouraged as an insecure practice, since it may allow malicious actors to have tokens or authorization codes sent to unexpected or attacker-controlled pages. Exercise caution if you decide to include a wildcard redirect URI in your configuration.
+
+    See the parameter [Details](https://developer.okta.com/docs/reference/api/apps/#details) section on the Apps API Reference page for configuration guidance on the use of wildcard subdomains.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-web-app/create-okta-application/index.md:22]: link should be shortened to (/docs/reference/api/apps/#details)

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 5]
```diff
@@ -1332,15 +1332,13 @@ Adds an OAuth 2.0 client application. This application is only available to the
 
 * At least one redirect URI and response type is required for all client types, with exceptions: if the client uses the [Resource Owner Password](https://tools.ietf.org/html/rfc6749#section-4.3) flow (if `grant_types` contains the value `password`) or [Client Credentials](https://tools.ietf.org/html/rfc6749#section-4.4) flow (if `grant_types` contains the value `client_credentials`) then no redirect URI or response type is necessary. In these cases you can pass either null or an empty array for these attributes.
 
-* If `wildcard_redirect` is `DISABLED`, all redirect URIs must be absolute URIs and must not include a fragment component.
+* If `wildcard_redirect` <ApiLifecycle access="ea" /> is `DISABLED`, all redirect URIs must be absolute URIs and must not include a fragment component. If `wildcard_redirect` is `SUBDOMAIN`, then configured redirect URIs may contain a single `*` character in the lowest-level domain to act as a wildcard. The wildcard subdomain must have at least one subdomain between it and the top level domain.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1335]: Do you want to use `mustn't` here? I'm always leery on contractions, so you're the arbiter.

Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1335]: then **(any/all)** configured redirect URIs...

Also, I' m not sure what a `lowest-level domain` might refer to - can we include an example like you have in the next bullet point?

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 6]
```diff
@@ -1332,15 +1332,13 @@ Adds an OAuth 2.0 client application. This application is only available to the
...
 
-* If `wildcard_redirect` is `SUBDOMAIN`, then configured redirect URIs may contain a single `*` character in the lowest-level domain to act as a wildcard. The wildcard subdomain must have at least one subdomain between it and the top level domain.
+  * The wildcard can match any valid hostname characters, but can't span more than one domain. As an example, if `https://redirect-*-domain.example.com/oidc/redirect` is configured as a redirect URI, then `https://redirect-1-domain.example.com/oidc/redirect` and `https://redirect-sub-domain.example.com/oidc/redirect` match, but `https://redirect-1.sub-domain.example.com/oidc/redirect` won't match.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1337]: *For* ~As an~ example, ...

Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1337]: **doesn't** match 
(the former is passive is it not?)

**Review by IanHakes-okta — APPROVED** [review]
Few comments - mostly nits, feel free to correct as you see fit and then advance past "Go!" and collect your $200.

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md (hunk 5 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1335]: Gah - I'd rather not - besides it MUST have not must NOT

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/create-okta-application/index.md (hunk 3 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/create-okta-application/index.md:23]: oops

**Pull request merged**